tests/test_async_client.py

```diff
@@ -16,6 +16,8 @@
 )
 async def test_publish_subscribe(qos_sub: QoS, qos_pub: QoS) -> None:
     async with AsyncMQTTClient() as client:
+        if qos_pub > client.maximum_qos:
+            return  # TODO add pytest.skip
```

Comment:
Why can't you use `pytest.skip()` right now? Under what circumstances would it currently skip the test?

Reply:
`pytest.skip` raises an exception which gets wrapped into an `ExceptionGroup`. Same problem as in #22 but I didn't want to replicate the solution from there here, only to then have you deal with a merge conflict.